docs/advanced/tls_porting.md

```diff
@@ -14,61 +14,83 @@ Creating session keys is only one use for random values; they have far more comp
 
 The preferred way to provide a custom entropy source:
 
-1. Implement the `mbedtls_hardware_poll` function to let it access the device's entropy source.
-2. Set the appropriate macros for your target in `hal/targets.json`.
+1. Implement the functions declared in `hal/trng_api.h` to let mbed TLS access the device's entropy source.
+2. Indicate that your target has an entropy source in `targets/targets.json`, by adding `TRNG` to your device's `device_has` section.
 
 The next two sections explain how to do this.
 
-## How to implement the hardware poll
+## How to implement the TRNG API
+
+The implementation of this interface has to be located in the mbed OS directory specific to your target. The name of this directory is of the form `targets/.../TARGET_<target name>`, for example in the case of K64F targets it is `targets/TARGET_Freescale/TARGET_KSDK2_MCUS/TARGET_MCU_K64F/`.
 
-The `mbedtls_hardware_poll` function is declared in `entropy_poll.h`:
+### Data structure
+
+You have to define a structure `trng_s` that holds all the information needed to operate the peripheral and describe its state.
+
+### Initialisation and release
+
+To enable initializing and releasing the peripherial, you must implement the following functions:
 
 ```C
-int mbedtls_hardware_poll( void *data,
-                           unsigned char *output, size_t len, size_t *olen );
+void trng_init(trng_t *obj);
+void trng_free(trng_t *obj);
 ```
 
-The implementation of this function needs to be in the HAL.
+### The entropy collector function
 
-### Parameters
+The function `trng_get_bytes()` serves as the primary interface to the entropy source. It is expected to load the collected entropy to the buffer and is declared as follows:
 
-- ``void *data``: A pointer to a structure containing the context to your entropy source. This parameter is `NULL` when the function is called by the framework. If you want to change this behavior, then please follow this [knowledge base article](https://tls.mbed.org/kb/how-to/add-entropy-sources-to-entropy-pool).
+```C
+int trng_get_bytes(trng_t *obj, uint8_t *output, size_t length, size_t *output_length);
+```
 
-- ``unsigned char *output``: A pointer to the output buffer. The function should write the entropy it collected to the buffer; mbed TLS then uses this data as entropy. Please consult your board's manual and write only the strongest entropy possible in this buffer. 
+- ``trng_t *obj``: `trng_t` is an alias to `trng_s` and it is the callers responsibility to initialise it before passing it to this function and deinitialise it (with the help of `trng_init()` and `trng_free()` respectively) when it is not required anymore.
```

Comment:
Please use international English i.e. `initialise` -> `initialize`. Also `deinitialise` -> `deinitialize`, but I am not sure that word exists.